src/modules/fancyzones/lib/Settings.cpp

```diff
@@ -32,8 +32,10 @@ struct FancyZonesSettings : winrt::implements<FancyZonesSettings, IFancyZonesSet
         PCWSTR name;
         bool* value;
         int resourceId;
-    } m_configBools[9] = {
+    } m_configBools[11] = {
+		{ L"fancyzones_enableOnInteract", &m_settings.enableOnInteract, IDS_SETTING_DESCRIPTION_ENABLEONINTERACT},

```

Comment:
`enableOnInteract` breaks the `shiftDrag` option :
 - when `enableOnInteract` and `shiftDrag` are both off, the zones don't activate (with or without pressing SHIFT).
 - when `enableOnInteract` is on and `shiftDrag` is off and , zones don't dis-activate (with or without pressing SHIFT).

Given the current behavior of `"On: Hold Shift key to enable zones while dragging"` isn't under design change review, changing it to `"On: Hold Shift key to interact with zones while dragging"` wouldn't be acceptable even if it wasn't causing the bug described above.

The general rules are:
 - if there are no community requests to change a behavior, don't change it (aka don't fix it if it's not broken)
 - if a change is proposed, it has to be simple to understand and not be a potential source of confusion: in this case having one option that basically reverse how other options are working is not ideal.

My proposal is to remove `enableOnInteract` and only add the `mouseDrag` option. If you want to make it a multiple choice use one of the existing controls for multiple choices (DropDown or ChoiceGroup).

Reply:
@enricogior

I debated this a lot myself. I'm happy to change it to the combination you'd prefer. I felt this gave the clearest UX, but I can appreciate not wanting to make those changes. So to be clear on the behavior:

For either mouse or shift, the expected behavior is:

|xDrag|xDown|zones visible?|note|
|-------|-------|---------------|-----|
|0|0|1|dragging enabled by default|
|0|1|0|actively disable dragging|
|1|0|0|dragging disabled by default|
|1|1|1|actively enable dragging|

So that is to say zones are enabled when xDrag XNOR xDown for whatever control. So what happens when these controls contradict one another? I.e. shiftDrag XNOR shiftDown != mouseDrag XNOR mouseDown. In this case, the behavior of the mouse may say to make zones enabled/disabled, while the behavior of shift says the opposite.

My suggestion to resolve this is to first bias in favor of the active choice (i.e. if a key is pressed, use that keys behavior and disregard the other regardless of setting). This resolves the bulk of the conflict scenarios, but it leaves us with two contradiction cases which are when both keys are active but the settings contradict:

|shiftDrag|mouseDrag|mouseDown|shiftDown|zones visible?|note|
|----------|-------------|--------------|-----------|---------------|-----|
|1|0|1|1|?|shift says disable, mouse says enable|
|0|1|1|1|?|shift says enable, mouse says disable|

I would suggest in these cases that zones be enabled. My reasoning is just a psychological one - if you're doing two things naively, you expect something to happen rather than nothing. 

Alternatively, we could give president to shift in these cases as it requires extra effort (the use of a second hand) and is less likely to be done accidentally. Do you have a preference?

I will fix the style errors once I'm clear on what intended behavior should be.

Reply:
A lot of good points.
I don't have a definitive answer, but I'm leaning towards the idea that we should have just one settings option for both Shift and Right-click.
So basically add some text to the current Shift settings saying `Hold Shift key or press left and right mouse buttons at the same time...`.
Then in the code you give precedence to the Shift key if both are the Shift key and the right button are pressed at the same time.
My only concern at the moment is if pressing both mouse buttons while dragging may break any app, in that case it would be useful to be able to disable it without disabling the Shift drag.
Also the other way around: if Shift drag breaks any app we should be able to disable it or allow to change the modifier.

I would suggest to give it a try to unified settings option, test it and decide if it's good as it is or if it will require changes.
Sounds good?